hbase-common/src/main/java/org/apache/hadoop/hbase/trace/TraceUtil.java

```diff
@@ -34,17 +51,24 @@
   private static HTraceConfiguration conf;
   private static Tracer tracer;
 
+  private static io.opentracing.Tracer otTracer;
+
+  private static final Logger LOG = LoggerFactory.getLogger(TraceUtil.class.getName());
+
+
   private TraceUtil() {
   }
 
-  public static void initTracer(Configuration c) {
-    if (c != null) {
-      conf = new HBaseHTraceConfiguration(c);
-    }
+  public static void initTracer(Configuration c, String serviceName) {
+    if (!GlobalTracer.isRegistered()) {
+      io.jaegertracing.Configuration conf = io.jaegertracing.Configuration.fromEnv(serviceName);
+      io.opentracing.Tracer tracer = conf.getTracerBuilder().build();
 
-    if (tracer == null && conf != null) {
-      tracer = new Tracer.Builder("Tracer").conf(conf).build();
+      GlobalTracer.register(tracer);
```

Comment:
More of a question  - This happens at the init time - means the server start up time right? If the user wants to use jaeger tracing he can do it by setting some env variables. But if the same has to be unset what needs to be done? And also after the server starts up and then he needs to introduce jaeger tracing how can that be done?

Reply:
I'd suggest you to browse the Jaeger's sampling doc: https://www.jaegertracing.io/docs/1.14/sampling/#collector-sampling-configuration
You can use env variable to control the sampling strategy. Or programmatically (at init). Finally, you can also define the sampling strategy at the central trace collector.